src/test/java/core/CredentialsTest.java

```diff
@@ -1,55 +1,227 @@
+        // verify credential is not present
+        verifyCredentialNotPresent(domainName, CRED_ID);
+    }
+
+    /**
+     * Verifies that a given input for a given credential id 
+     * is equals to the value expected. If there are two credentials
+     * with the same id (that should not happen), the first one is verified.
+     * 
+     * @param credId
+     * @param inputId
+     * @param expected
+     */
+    private void verifyInputValue(String credId, String inputId, String expected) {
+        jenkins.visit("credentials");
+        // Find credential div
+        String actual = findIfNotVisible(by.input(credId))
+                        .findElement(by.ancestor("div"))
+                        // Find input to verify inside credential div
+                        .findElement(by.input(inputId))
+                        .getAttribute("value");
+        assertThat(actual, equalTo(expected));
+    }
+    
+    /**
+     * Verifies that a given credential is not present in the entire credentials page
+     * 
+     * @param credId
+     */
+    private void verifyCredentialNotPresent(String credId) {
+        assertThat(getElement(by.input(credId)), nullValue());        
+    }
+    
+    /**
+     * Verifies that a given input for a given credential id 
+     * is equals to the value expected under the scope of a given domain.
+     * If there are two credentials with the same id in the same domain,
+     * the first one is verified.
+     *
+     * @param domainName
+     * @param credId
+     * @param inputId
+     * @param expected
+     */
+    private void verifyInputValueInDomain(String domainName, String credId, String inputId, String expected) {
+        jenkins.visit("credentials");
+        
+        // Find domain div
+        String actual = findIfNotVisible(by.input(domainName))
+                        .findElement(by.ancestor("div"))
+                         // Find credential div inside domain div
+                        .findElement(by.input(credId))
+                        .findElement(by.ancestor("div"))
+                         // Find input with inputId passed
+                        .findElement(by.input(inputId))
+                        .getAttribute("value");
+        
+        assertThat(actual, equalTo(expected));
+    }
+    
+    /**
+     * Verifies that a given credential is not present under the scope of a given domain.
+     * 
+     * @param domainName
+     * @param credId
+     */
+    private void verifyCredentialNotPresent(String domainName, String credId) {
+       try {
+           // Find domain div
+            findIfNotVisible(by.input(domainName))
+           .findElement(by.ancestor("div"))
+            // Find credential div inside domain div
+           .findElement(by.input(credId))
+           .findElement(by.ancestor("div"))
+           // Find input with credential id passed as parameter
+           .findElement(by.input(credId));
+       } catch (NoSuchElementException e) {
```

Comment:
My first idea was to actually use it. I discarded that solution because it would involve requiring another plugin that uses credentials for that and I did not want to add another variable to the test case, that is, adding some other failure point not directly related with the credentials plugin. I could do it though, for example, require the Git plugin and when creating a FreeStyleProject, checking that the credentials created appear in the credentials dropdown. What do you think?